src/views/Portfolio/Portfolio.tsx

```diff
@@ -1,16 +1,35 @@
-import React, { memo } from "react";
-import { Typography } from "@mui/material";
+import React, { memo, useCallback, useEffect, useState } from "react";
 import Page from "components/Page";
 import Drawer from "../../components/Drawer";
-import SearchBar from "components/SearchBar";
+import WidgetContainer from "views/Dashboard/components/WidgetContainer";
+import useBreakpoint from "hooks/useBreakpoint";
+import JUPAppBar from "components/JUPAppBar";
+import PortfolioApp from "./components/Widgets/PortfolioApp";
 
 const Portfolio: React.FC = () => {
+  const [drawerIsOpen, setDrawerIsOpen] = useState<boolean>(true);
```

Comment:
not a big fan of this being at this level. obviously it works, but i'm seeing it's needing to be put into each page. i'd expect it to be at a higher level and handled there. and then if necessary, pass this state value of the drawer isOpen down into each of the views as a prop.

i hope this makes sense. low priority but will dry up the code more as you go through the next few pages.

Reply:
yep, i already identified it and I've tried to refactor it a couple of times but I can't get it working at the page level yet